src/conn/conn_compact.c

```diff
@@ -18,6 +21,61 @@ __compact_server_run_chk(WT_SESSION_IMPL *session)
     return (FLD_ISSET(S2C(session)->server_flags, WT_CONN_SERVER_COMPACT));
 }
 
+/*
+ * __background_compact_find_next_uri --
+ *     Given a URI, find the next one in the metadata file that is eligible for compaction.
+ */
+static int
+__background_compact_find_next_uri(WT_SESSION_IMPL *session, WT_ITEM *uri, WT_ITEM *next_uri)
+{
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    int exact;
+    const char *key;
+
+    cursor = NULL;
+    exact = 0;
+    key = NULL;
+
+    /* Use a metadata cursor to have access to the existing URIs. */
+    WT_ERR(__wt_metadata_cursor(session, &cursor));
+
+    /* Position the cursor on the given URI. */
+    cursor->set_key(cursor, (const char *)uri->data);
+    WT_ERR(cursor->search_near(cursor, &exact));
+
+    /*
+     * The given URI may not exist in the metadata file. Since we always want to return a URI that
+     * is lexicographically larger the given one, make sure not to go backwards.
+     */
+    if (exact <= 0)
+        WT_ERR(cursor->next(cursor));
+
+    /* Loop through the eligible candidates. */
+    while (ret == 0) {
+        WT_ERR(cursor->get_key(cursor, &key));
+        /* Check we are still dealing with keys which have the right prefix. */
+        if (!WT_PREFIX_MATCH(key, WT_COMPACT_URI_PREFIX)) {
+            ret = WT_NOTFOUND;
+            break;
+        }
+        /* There are files that should not be compacted. */
+        if (!WT_STREQ(key, WT_HS_URI))
+            /* FIXME-WT-11343: check if the table is supposed to be compacted. */
+            break;
```

Comment:
Shouldn't this cause the loop to exit or simply skip that entry?

Reply:
If the key **is not** equal to `WT_HS_URI`, we want to break from the loop as this is a key we can process. If this is the HS store file, we need to call `next`.